### PR TITLE
[WIP] reduce package dependency (fixes #5)

### LINKED ISSUE #5: Cut down on the dependencies
According to JuliaObsverver 
ReferenceTests.jl has 55 dependencies (5 direct).

For comparison: TensorFlow.jl has 30 (13 direct).
DifferentialEquations has 84 (26 direct).

It would be really nice not to be haulling down the whole Images ecosystem when doing a reference test on plain text.

I'm not sure what the current state of the optional dependencies setup is.
I'm not sure how happy [Requires.jl](https://github.com/MikeInnes/Requires.jl) is with precompilation etc,
or what is going on in the Pkg3 space with regards to it.

But anyway, it is rather annoying that every time I run a test huge amounts of time is spent cloning then removing packages.
Because I tend to list ReferenceTests.jl as a Test time dependency only.
(On my local boxes I really need to remember to install it)

### PR DESCRIPTION
**Pull request by johnnychen94**
Two things are done here:

- [x] upgrade to the new registrator and drop 0.7 compatibility
- [x] reduce package dependency #5 #11 

After this PR you need to:

* install https://github.com/JuliaComputing/Registrator.jl
* [optional] install https://github.com/christopher-dG/tag-bot
* bump a version

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name = "ReferenceTests"
+uuid = "324d217c-45ce-50fc-942e-d289b448e8cf"
+authors = ["Evizero <stocker.christof@gmail.com>"]
```
Comment (Evizero) [inline Project.toml]: Given all the changes he made I'd like to add @oxinabox to the list of authors

```suggestion
authors = ["Christof Stocker <stocker.christof@gmail.com>", "Lyndon White <oxinabox@ucc.asn.au>"]
```

**Comment by johnnychen94** [comment]
Benchmark:

```bash
jc@mathai0:~/Documents/ReferenceTests.jl$ time julia -e "using ReferenceTests"

real	0m7.981s
user	0m7.870s
sys	0m0.844s
jc@mathai0:~/Documents/ReferenceTests.jl$ time julia -e 'include("src/ReferenceTests.jl")'

real	0m1.093s
user	0m1.221s
sys	0m0.604s
```

File: src/ReferenceTests.jl [hunk 2]
```diff
@@ -17,6 +16,19 @@ export
...
+include("handlers/text.jl")
+
+function __init__()
+    # images type
+    @require ColorTypes="3da002f7-5984-5a60-b8a6-cbb66c0b333f" include("handlers/image.jl")
+    @require Colors="5ae59095-9a9b-59fe-a467-6f913c188581" include("handlers/image.jl")
+    @require ImageCore="a09fc81d-aa75-5fe9-8630-4744c3626534" include("handlers/image.jl")
+    @require ImageInTerminal="d8c32880-2388-543b-8c61-d9f865259254" include("handlers/image.jl")
+    @require Images="916415d5-f1e6-5110-898d-aaa5f9f070e0" include("handlers/image.jl")
+    @require TestImages="5e47fb64-e119-507b-a336-dd2b206d9990" include("handlers/image.jl")
```
Comment (Evizero) [inline src/ReferenceTests.jl]: i don't know the correct way, but this looks wrong (including the same file multiple times). maybe Lyndon has an idea

**Comment by johnnychen94** [comment]
The most appropriate place for `@test_approx_eq_sigma_eps` is `ImageDistances.jl`, let's see if I can make it ported this week.

**Comment by oxinabox** [comment]
I hear Requires.jl is mad slow in julia 1.x

File: src/ReferenceTests.jl (hunk 2 above)
Comment (oxinabox) [reply src/ReferenceTests.jl]: This looks wrong to me also.
Won't that result in it being included if just one of the multipled dependencies is loaded?

**Comment by Evizero** [comment]
> I hear Requires.jl is mad slow in julia 1.x

Interessting. Maybe lets just do the registrator update first and leave the dependency reduction for another time/pr ?

**Comment by johnnychen94** [comment]
🤔 perhaps remove the dependency on `Images` is sufficient

**Comment by johnnychen94** [comment]
reopen it if there's a need.

**Pull request closed without merging**